router/internal/jitterticker/jitterticker.go

```diff
@@ -0,0 +1,64 @@
+package jitterticker
+
+import (
+	"math/rand"
+	"time"
+)
+
+// Ticker is a time.Ticker-like struct that adds jitter to the interval
+type Ticker struct {
+	interval time.Duration
+
+	maxJitter time.Duration
+
+	C    <-chan time.Time
+	stop chan struct{}
+}
+
+func NewTicker(interval time.Duration, maxJitter time.Duration) *Ticker {
+	if interval < 0 {
+		panic("negative interval")
+	}
+
+	if maxJitter < 0 {
+		panic("negative max jitter")
+	}
+
+	c := make(chan time.Time)
+	stop := make(chan struct{})
+
+	ticker := &Ticker{
+		C:         c,
+		stop:      stop,
+		interval:  interval,
+		maxJitter: maxJitter,
+	}
+
+	go func() {
+		for {
+			time.Sleep(ticker.getDelay())
+
+			select {
+			case <-ticker.stop:
+				close(c)
+				return
+			case c <- time.Now():
+			default:
```

Comment:
We need to be careful here. For with a default can lead to extensive CPU usage. Unlikely, but preventable by removing the default statement. The rest is optimized by go runtime.